scrapper.py

```diff
@@ -44,20 +78,106 @@ def get_search_urls(self):
         pass
 
 
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def _fill_article_with_meta_information(self, article_bs):
+        # DATE
+        # the second date - of publication
+        datetime_of_publication = article_bs.find('span', class_='echo_date')['data-published']
+        self.article.date = datetime.strptime(datetime_of_publication, '%Y-%m-%dT%H:%M:%S%z')
+
+        # AUTHOR
+        try:
+            self.article.author = article_bs.find('div', class_='meta-item meta-item_author').text.strip()
+        except AttributeError:
+            self.article.author = 'NOT FOUND'
+        # TOPICS
+        self.article.topics = 'NOT FOUND'
+
+        # TITLE
+        self.article.title = article_bs.find('div', class_="post-title")
+        if self.article.title is not None:
+            self.article.title = self.article.title.text.splitlines()[1]
+
+    def _fill_article_with_text(self, article_bs):
+        # only text of the article (lead + body)
+        self.article.text = article_bs.find('div', class_='post-lead').text.strip()
+        self.article.text += article_bs.find('div', class_='body').text.strip()
+
+    def parse(self):
+        sleep(random.randint(1, 5))
+        response = requests.get(url=self.article_url, headers={
+            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36'
+                          '(KHTML, like Gecko) Chrome/98.0.4758.109 Safari/537.36'
+        })
+        if not response.ok:
+            return None
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    # path to folder
+    path_for_environment = Path(base_path)
+
+    if path_for_environment.exists():
+        shutil.rmtree(path_for_environment)
+    path_for_environment.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path) as file:
+        scrapper_config = json.load(file)
 
+    for url in scrapper_config["seed_urls"]:
```

Comment:
what if `scrapper_config["seed_urls"]` is not iterable? TypeError will be thrown before IncorrectURLError